File: BikeCheckAndroid/app/src/main/java/com/bikecheck/android/ui/onboarding/OnboardingViewModel.kt
package com.bikecheck.android.ui.onboarding

import android.content.SharedPreferences
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.bikecheck.android.data.repository.StravaRepository
import com.bikecheck.android.models.TourStep
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.launch
import javax.inject.Inject

@HiltViewModel
class OnboardingViewModel @Inject constructor(
    private val stravaRepository: StravaRepository,
    private val sharedPreferences: SharedPreferences
) : ViewModel() {
    
    private val _showOnboarding = MutableStateFlow(false)
    val showOnboarding: StateFlow<Boolean> = _showOnboarding
    
    private val _showTour = MutableStateFlow(false)
    val showTour: StateFlow<Boolean> = _showTour
    
    private val _currentTourStep = MutableStateFlow(0)
    val currentTourStep: StateFlow<Int> = _currentTourStep
    
    private val _isLoadingTestData = MutableStateFlow(false)
    val isLoadingTestData: StateFlow<Boolean> = _isLoadingTestData
    
    fun startOnboarding() {
        _showOnboarding.value = true
    }
    
    fun loadTestDataIfNeeded() {
        if (_isLoadingTestData.value) return
        
        _isLoadingTestData.value = true
        viewModelScope.launch {
            try {
                stravaRepository.insertTestData()
            } finally {
                _isLoadingTestData.value = false
            }
        }
    }
    
    fun startTour() {
        _showOnboarding.value = false
        _showTour.value = true
        _currentTourStep.value = 0
        
        // Load test data when starting tour
        loadTestDataIfNeeded()
    }
    
    fun skipTour() {
        _showOnboarding.value = false
        _showTour.value = false
        markOnboardingComplete()
        clearTestData()
    }
    
    fun nextTourStep() {
        val tourSteps = TourStep.values()
        if (_currentTourStep.value < tourSteps.size - 1) {
            _currentTourStep.value += 1
        } else {
            completeTour()
        }
    }
    
    fun previousTourStep() {
        if (_currentTourStep.value > 0) {
            _currentTourStep.value -= 1
        }
    }
    
    fun completeTour() {
        _showTour.value = false
        _currentTourStep.value = 0
        
        // Mark onboarding as complete
        markOnboardingComplete()
        
        // Clear test data when tour completes
        clearTestData()
    }
    
    private fun markOnboardingComplete() {
        sharedPreferences.edit()
            .putBoolean("is_first_launch", false)
            .apply()
    }
    
    private fun clearTestData() {
        viewModelScope.launch {
            // In a full implementation, you would clear test data here
            // For now, we'll leave the test data in place
        }
    }
    
    fun getCurrentTourStep(): TourStep? {
        val tourSteps = TourStep.values()
        return if (_currentTourStep.value < tourSteps.size) {
            tourSteps[_currentTourStep.value]
        } else {
            null
        }
    }
}

File: bikecheck/ViewModels/OnboardingViewModel.swift
import Foundation
import Combine

class OnboardingViewModel: ObservableObject {
    @Published var showOnboarding: Bool = false
    @Published var showTour: Bool = false
    @Published var currentTourStep: Int = 0
    @Published var isLoadingTestData: Bool = false
    
    private let stravaService = StravaService.shared
    private var cancellables = Set<AnyCancellable>()
    
    init() {}
    
    func startOnboarding() {
        showOnboarding = true
    }
    
    func loadTestDataIfNeeded() {
        guard !isLoadingTestData else { return }
        
        isLoadingTestData = true
        
        DispatchQueue.global(qos: .userInitiated).async {
            // Load test data but don't auto-sign in during onboarding
            self.stravaService.insertTestData(autoSignIn: false)
            
            DispatchQueue.main.async {
                self.isLoadingTestData = false
            }
        }
    }
    
    func startTour() {
        showOnboarding = false
        showTour = true
        currentTourStep = 0

        // Load test data and sign in when starting tour
        loadTestDataIfNeeded()
        DispatchQueue.main.async {
            self.stravaService.isSignedIn = true
        }
    }
    
    func skipTour() {
        showOnboarding = false
        showTour = false
        // Don't clear test data - user is already logged in
        // Just dismiss the onboarding overlay
    }
    
    func nextTourStep() {
        let tourSteps = TourStep.allCases
        if currentTourStep < tourSteps.count - 1 {
            currentTourStep += 1
        } else {
            completeTour()
        }
    }
    
    func completeTour() {
        showTour = false
        currentTourStep = 0
        // Don't clear test data - user is already logged in
        // Just dismiss the tour overlay
    }
    
    private func clearTestData() {
        DispatchQueue.global(qos: .userInitiated).async {
            self.stravaService.clearTestData()
        }
    }
    
    func clearTestDataIfNeeded() {
        // Clear test data and ensure user is signed out when returning to login
        DispatchQueue.main.async {
            self.stravaService.isSignedIn = false
        }
        clearTestData()
    }
    
    func getCurrentTourStep() -> TourStep? {
        let tourSteps = TourStep.allCases
        guard currentTourStep < tourSteps.count else { return nil }
        return tourSteps[currentTourStep]
    }
}